app/models/discovery_rule.rb

```diff
@@ -37,4 +38,16 @@ def default_int_attributes
    self.max_count ||= 0
    self.priority  ||= 0
   end
+
+  def enforce_taxonomy
+    return if hostgroup.nil?
+    unless (ms = hostgroup.organizations - organizations).empty?
+      names = ms.collect(&:name).inspect
```

Comment:
Prefer .to_sentence over inspect, it's more user-friendly - or leave the names out.